### PR TITLE
Charger Crusher Return

### PR DESCRIPTION
**Pull request by PhantomEpicness**
Due to github fuckery, I was forced to make a new PR

old PR: https://github.com/cmss13-devs/cmss13/pull/14
## About The Pull Request

Old momentum crusher has returned in form of the "Charger" crusher. The basis is simple, hold move key and destroy shit.

Passive: Directional armor. Charger has 50 frontal armor, 35 side armor, and 25 rear armor. Because of this ability, charger crusher is unable to direction lock itself.

Active: Charge
Enables charging in which you will become a living train and ram shit at the cost of being plasma heavy. Upon reaching maximum momentum, you will roar and deal full damage.

Active: Tumble
 Clicking to the side of where the crusher is facing will launch it 2 tiles to the side in the clicked direction. Hitting a human will knock them over. This ability can be used while charge to quickly offset yourself. However, it will reset your momentum to 0.

Active: Ram

An ability that allows you to fling a person in front of you for a short distance. Has a 2 second stun and a longer 10 second cooldown. Subject to change and/or removal.

Active: Charger stomp. Unlike the base crusher, it does not stun, but does heavy damage to any human unfortunate enough to be under the charger, it also applies a mild screen shake to any humans in view and can be used to help escape.

Credit to original charger devs, lumdor and donk!

## Why It's Good For The Game

design doc: https://github.com/cmss13-devs/cmss13-design-docs/pull/9

_tl:dr_ 
Charger breathes new life into normal crusher by bringing back its most requested features pre-moba features in form of a strain. The caste design is much more simple compared to other xeno castes. One simple, yet tried and true fun combo exists, charging.
## Changelog

:cl: TotalEpicness, TheDonkified and Lumdor
add: Added charger strain for crusher.
experiment: Charger functions very similarly to old crusher. Hold movement key and after a while you gain momentum to ram humans and certain objects. Behavior upon collision with different things are subject to change.
Passive: Directional armor. Charger has 50 frontal armor, 35 side armor, and 25 rear armor. Because of this ability, charger crusher is unable to direction lock itself.
experiment: Active 1: Toggle Charging. Use this ability to toggle charging on and off.
experiment: Active 2: Tumble. Clicking to the side of where the crusher is facing will launch it 2 tiles to the side in the clicked direction. Hitting a human will knock them over. This ability can be used while charge to quickly offset yourself. However, it will reset your momentum to 0.
add: Tumble now has cool sound effects and a spin.
sprite: Added tumble ability sprites
tweak: Charger now takes 3 full hits to destroy a normal wall instead of destroying it in pretty much one. Xeno walls take half damage
experiment: Active 3: Charger stomp. Unlike the base crusher, it does not stun, but does heavy damage to any human unfortunate enough to be under the charger, it also applies a mild screen shake to any humans in view and can be used to help escape.
experiment: Active 4: Ram. A simple offensive ability that flings mobs without stun. Much like warrior fling.
add: Xenos blocking a charger's path when charger get pushed to the side, if no clear path to move to exists, they get flicker stunned so that charger may charge.
/:cl:

### CODE REVIEW

**Comment by PhantomEpicness** [comment]
Charger changelog
- return of base Crusher
- bonus charging armor removal
- Ram is now a 2 tile ranged warrior fling with a 2 second stun. No more mini base charge.
- fixed macro bugs, if you think macros should be rebinded, let me know
- slight health nerf and frontal armor bonus is now 25 
I am also currently working on a small shield bonus that your recieve when hitting max momentum that lasts a few seconds. This is a solution that will give charger a bit more survivability and slightly lower the risk of charging without making it excessively tanky like it was previously.
this won't make it to this TM however, as I need to see if it's even needed.

**Comment by novaepee** [comment]
I can ensure you that all marines shall hate Ram. Over at TGMC, crusher has a similar ability, and good crusher players fish new marine players out of barricades.

**Comment by Chainsawicus** [comment]
> A small shield bonus that you recieve when hitting max momentum that lasts a few seconds. This solution will give charger a bit more survivability and slightly lower the risk of charging without making it excessively tanky like it was previously. this won't make it to this TM however, as I need to see if it's even required.

No more shields... please...

In seriousness, since shields as a whole are being toned down or flat out removed in certain cases soon, I think you should look to alternative methods of survivability, as there's no guarantee that charger would be able to keep its shields.

**Comment by PhantomEpicness** [comment]
> > A small shield bonus that you recieve when hitting max momentum that lasts a few seconds. This solution will give charger a bit more survivability and slightly lower the risk of charging without making it excessively tanky like it was previously. this won't make it to this TM however, as I need to see if it's even required.
> 
> No more shields... please...
> 
> In seriousness, since shields as a whole are being toned down or flat out removed in certain cases soon, I think you should look to alternative methods of survivability, as there's no guarantee that charger would be able to keep its shields.

Yeah now that it was brought up to me that shields were being done away with, I have considered other options.

**Comment by stanalbatross** [comment]
image
fix

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 1]
```diff
@@ -307,24 +310,58 @@ In most cases it makes more sense to use apply_damage() instead! And make sure t
...
+	var/armour_config = GLOB.marine_ranged
+	if(armour_type == ARMOR_MELEE)
+		armour_config = GLOB.marine_melee
+
+	var/modified_damage = armor_damage_reduction(armour_config, damage, armor, penetration, 0, 0)
+	apply_damage(modified_damage, damage_type, target_limb)
+
+	return modified_damage
+
+
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/human/human_damage.dm:363]: remove commented out code all over the pr once you're done with the final implementation

**Comment by PhantomEpicness** [comment]
Changes:

- frenzy NO LONGER affects charger at all. 
- fixed macros, they should be assigned to correct keys in order now
- 5 less back armor. Note that frontal and side armor has been changed to compensate for 5 less base armor, effectively only lowering back armor.
- hp increase 760 > 780
- fling stun removed
- fling distance 2 > 3
- changed description to a more accurate one:

> Your charge is now momentum based, the further you go, the more damage and speed you will get until you achieve max momentum when you roar. Your armor is now directional, being the toughest on the front, weaker on the sides and weakest at the back. Your shield is also removed. In return you gain more health, gain an ability to tumble to avoid enemies, and gain an ability to forcefully move enemies via ramming. Finally, you trade being able to resist slowdowns from tall autospitters for being unaffected by frenzy pheros.

**Comment by stanalbatross** [comment]
fine in its current implementation just needs code review

**Comment by Carnage068** [comment]
let's gooooo

File: code/game/objects/structures/crates_lockers/closets.dm [hunk 2]
```diff
@@ -154,6 +154,15 @@
 		to_chat(user, SPAN_NOTICE("It won't budge!"))
 	return
 
+
+/obj/structure/closet/proc/take_damage(damage)
+	health = max(health - damage, 0)
+	if(health == 0)
```
Comment (stanalbatross) [inline code/game/objects/structures/crates_lockers/closets.dm:160]:
```suggestion
	if(health <= 0)
```
sanity

File: code/game/objects/structures/girders.dm [hunk 3]
```diff
@@ -310,11 +310,18 @@
...
 		health -= dmg
+		take_damage(dmg)
 		bullet_ping(P)
 	if(health <= 0)
 		update_state()
 	return TRUE
 
+/obj/structure/girder/proc/take_damage(damage)
+	health = max(health - damage, 0)
+	if(health == 0)
```
Comment (stanalbatross) [inline code/game/objects/structures/girders.dm:321]:
```suggestion
	if(health <= 0)
```

File: code/modules/mob/living/carbon/human/human_damage.dm [hunk 4]
```diff
@@ -307,24 +310,58 @@ In most cases it makes more sense to use apply_damage() instead! And make sure t
...
+*/
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/human/human_damage.dm:364]: remove

File: code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_abilities.dm [hunk 5]
```diff
@@ -51,14 +48,221 @@
...
+	name = "Tumble"
+	action_icon_state = "tumble"
+	macro_path = /datum/action/xeno_action/verb/verb_crusher_tumble
+	action_type = XENO_ACTION_CLICK
+	ability_primacy = XENO_PRIMARY_ACTION_2
+
+	plasma_cost = 25
+	xeno_cooldown = 10 SECONDS
+
+/datum/action/xeno_action/activable/tumble/proc/handle_mob_collision(mob/living/carbon/human/Human)
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_abilities.dm:250]: need a sanity check to see if the arg being parsed actually is a human

File: code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_powers.dm [hunk 6]
```diff
@@ -107,7 +107,58 @@
 	..()
 	return
 
-/datum/action/xeno_action/onclick/crusher_shield/use_ability(atom/A)
+/datum/action/xeno_action/onclick/crusher_stomp/charger/use_ability(atom/Target)
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_powers.dm:110]: needs a target sanity check

File: code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_powers.dm [hunk 7]
```diff
@@ -107,7 +107,58 @@
...
+	playsound(get_turf(Xeno), 'sound/effects/bang.ogg', 25, 0)
+	Xeno.visible_message(SPAN_XENODANGER("[Xeno] smashes into the ground!"), SPAN_XENODANGER("You smash into the ground!"))
+	Xeno.create_stomp()
+
+	for (var/mob/living/carbon/Human in get_turf(Xeno)) // MOBS ONTOP
+		if (Human.stat == DEAD || Xeno.can_not_harm(Human))
+			continue
+
+		new effect_type_base(Human, Xeno, , , get_xeno_stun_duration(Human, effect_duration))
+		to_chat(Human, SPAN_XENOHIGHDANGER("You are BRUTALLY crushed an stompted on by [Xeno] !!!"))
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_powers.dm:134]:
```suggestion
		to_chat(Human, SPAN_XENOHIGHDANGER("You are BRUTALLY crushed and stomped on by [Xeno]!!!"))
```

File: code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_powers.dm [hunk 8]
```diff
@@ -107,7 +107,58 @@
...
+		Targeted = Human
+	for (var/mob/living/carbon/Human in orange(distance, get_turf(Xeno))) // MOBS AROUND
+		if (Human.stat == DEAD || Xeno.can_not_harm(Human))
+			continue
+		//new effect_type_base(H, X, , , get_xeno_stun_duration(H, effect_duration))
+		//if(H.mob_size < MOB_SIZE_BIG)
+			//H.KnockDown(get_xeno_stun_duration(H, 0.2))
+		if(Human.client)
+			shake_camera(Human, 2, 2)
+		//to_chat(H, SPAN_XENOHIGHDANGER("You are slowed as [X] knocks you off balance!"))
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_powers.dm:152]: zoink this commented stuff

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 9]
```diff
@@ -0,0 +1,581 @@
...
+	if (. == 0)
+		return
+
+	var/mob/living/carbon/Xenomorph/Crusher/C = MS.xeno
+	C.mutation_type = CRUSHER_CHARGER
+	C.small_explosives_stun = FALSE
+	C.health_modifier += XENO_HEALTH_MOD_LARGE
+	C.speed_modifier += XENO_SPEED_FASTMOD_TIER_3
+	C.armor_modifier -= XENO_ARMOR_MOD_SMALL
+	C.ignore_aura = "frenzy" // no funny crushers going 7 morbillion kilometers per second
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:33]: make this known to the crushers

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 10]
```diff
@@ -0,0 +1,581 @@
...
+/obj/structure/window/handle_charge_collision(mob/living/carbon/Xenomorph/X, datum/action/xeno_action/onclick/charger_charge/CCA)
+	if(unacidable)
+		CCA.stop_momentum()
+		return
+
+	if(!CCA.momentum)
+		CCA.stop_momentum()
+		return
+
+	health -= CCA.momentum * 40 //Usually knocks it down.
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:88]: use a define

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 11]
```diff
@@ -0,0 +1,581 @@
...
+/obj/structure/grille/handle_charge_collision(mob/living/carbon/Xenomorph/X, datum/action/xeno_action/onclick/charger_charge/CCA)
+	if(unacidable)
+		CCA.stop_momentum()
+		return
+
+	if(!CCA.momentum)
+		CCA.stop_momentum()
+		return
+
+	health -= CCA.momentum * 40 //Usually knocks it down.
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:108]: as above

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 12]
```diff
@@ -0,0 +1,581 @@
...
+	healthcheck()
+
+	if(QDELETED(src))
+		CCA.lose_momentum(1) //Lose one turf worth of speed
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:112]: use defines for these loss amounts

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 13]
```diff
@@ -0,0 +1,581 @@
...
+// Vending machines
+
+/obj/structure/machinery/vending/handle_charge_collision(mob/living/carbon/Xenomorph/X, datum/action/xeno_action/onclick/charger_charge/CCA)
+	if(CCA.momentum >= 3)
+		if(unacidable)
+			CCA.stop_momentum()
+			return
+		X.visible_message(
+			SPAN_DANGER("[X] smashes straight into [src]!"),
+			SPAN_XENODANGER("You smash straight into [src]!")
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:141]:
```suggestion
			SPAN_DANGER("[X] smashes straight into \the [src]!"),
			SPAN_XENODANGER("You smash straight into \the [src]!")
```

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 14]
```diff
@@ -0,0 +1,581 @@
...
+	CCA.stop_momentum()
+
+
+// Barine Vending machines
+
+/obj/structure/machinery/cm_vending/handle_charge_collision(mob/living/carbon/Xenomorph/X, datum/action/xeno_action/onclick/charger_charge/CCA)
+	if(CCA.momentum >= 3)
+		X.visible_message(
+			SPAN_DANGER("[X] smashes straight into [src]!"),
+			SPAN_XENODANGER("You smash straight into [src]!")
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:159]: \the

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 15]
```diff
@@ -0,0 +1,581 @@
...
+	Crossed(X)
+	return XENO_CHARGE_TRY_MOVE
+
+// Cades
+
+/obj/structure/barricade/handle_charge_collision(mob/living/carbon/Xenomorph/X, datum/action/xeno_action/onclick/charger_charge/CCA)
+	if(CCA.momentum)
+		visible_message(
+			SPAN_DANGER("[X] rams into [src] and skids to a halt!"),
+			SPAN_XENOWARNING("You ram into [src] and skid to a halt!")
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:192]: \the

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 16]
```diff
@@ -0,0 +1,581 @@
...
+
+/obj/structure/machinery/defenses/sentry/handle_charge_collision(mob/living/carbon/Xenomorph/X, datum/action/xeno_action/onclick/charger_charge/CCA)
+	if(!CCA.momentum)
+		CCA.stop_momentum()
+		return
+	var/datum/effect_system/spark_spread/s = new
+	s.set_up(5, 1, loc)
+	X.visible_message(
+		SPAN_DANGER("[X] rams [src]!"),
+		SPAN_XENODANGER("You ram [src]!")
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:471]: \the

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 17]
```diff
@@ -0,0 +1,581 @@
...
+	if(CCA.momentum > 1)
+		CrusherImpact()
+		var/datum/effect_system/spark_spread/s = new
+		update_health(CCA.momentum * 15)
+		if(operator)	operator.emote("pain")
+		s.set_up(1, 1, loc)
+		s.start()
+		X.visible_message(
+			SPAN_DANGER("[X] rams [src]!"),
+			SPAN_XENODANGER("You ram [src]!")
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:497]: \the

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 18]
```diff
@@ -0,0 +1,581 @@
...
+		)
+		playsound(src, "sound/effects/metal_crash.ogg", 25, TRUE)
+		if(istype(src,/obj/structure/machinery/m56d_hmg/auto)) // we don't want to charge it to the point of downgrading it (:
+			var/obj/item/device/m2c_gun/HMG = new(src.loc)
+			to_chat(world, "TEST STEST")
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:502]: 💀

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 19]
```diff
@@ -0,0 +1,581 @@
...
+
+// Filing Cabinets
+
+/obj/structure/filingcabinet/handle_charge_collision(mob/living/carbon/Xenomorph/X, datum/action/xeno_action/onclick/charger_charge/CCA)
+	if(!CCA.momentum)
+		CCA.stop_momentum()
+		return
+	X.visible_message(
+		SPAN_DANGER("[X] rams [src]!"),
+		SPAN_XENODANGER("You ram [src]!")
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:553]: \the

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm [hunk 20]
```diff
@@ -0,0 +1,581 @@
...
+// Todo: Give this and other shitty fucking indestructable legacy items proper destruction mechanics. This includes being vunerable to bullets,explosions, etc and not just the charger.
+// For now this is fine since priority is charger, and I'm not willing to spend all day looking for bumfuck legacy item #382321 thats used a total of three times in the entireity of CM and adding health and everything to it.
+
+/obj/structure/dispenser/handle_charge_collision(mob/living/carbon/Xenomorph/X, datum/action/xeno_action/onclick/charger_charge/CCA)
+	if(!CCA.momentum)
+		CCA.stop_momentum()
+		return
+	X.visible_message(
+		SPAN_DANGER("[X] rams [src]!"),
+		SPAN_XENODANGER("You ram [src]!")
```
Comment (stanalbatross) [inline code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:573]: \the

**Review by stanalbatross — CHANGES_REQUESTED** [review]

File: code/modules/mob/living/carbon/human/human_damage.dm (hunk 4 above)
Comment (PhantomEpicness) [reply code/modules/mob/living/carbon/human/human_damage.dm:364]: done

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm (hunk 9 above)
Comment (PhantomEpicness) [reply code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:33]: stated in strain desc

**Comment by stanalbatross** [comment]
fix conflicts + do suggestions

**Comment by Carnage068** [comment]
we back in this

File: code/modules/defenses/bell_tower.dm [hunk 21]
```diff
@@ -139,7 +142,8 @@
 
 /obj/item/device/motiondetector/internal/apply_debuff(mob/target)
 	var/mob/living/to_apply = target
-
+	if(HAS_TRAIT(to_apply, TRAIT_CHARGING))
+		to_chat(to_apply, SPAN_WARNING("You ignore some weird noises as you charge."))
```
Comment (stanalbatross) [inline code/modules/defenses/bell_tower.dm:146]: return under here

File: code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_abilities.dm (hunk 5 above)
Comment (stanalbatross) [reply code/modules/mob/living/carbon/xenomorph/abilities/crusher/crusher_abilities.dm:250]: do

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm (hunk 13 above)
Comment (stanalbatross) [reply code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:141]: missed

File: code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm (hunk 15 above)
Comment (stanalbatross) [reply code/modules/mob/living/carbon/xenomorph/mutators/strains/crusher/charger.dm:192]: missed

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by PhantomEpicness** [comment]
Issue:

Alacrit fruit makes crusher go frenzy speed

**Comment by Misttyy** [comment]
I really like charger except for one thing.

I've noticed a bit, and talked with other marine players to confirm it wasn't a one off thing. One charger ram can give someone 7-8 bleeds, or make them bleed across their entire body. This feels very excessive.

**Review by stanalbatross — APPROVED** [review]
any further issues with alacrit can be looked at with balance to the fruit itself

**Comment by PhantomEpicness** [comment]
Made alacrit fruit work 

> I really like charger except for one thing.
> 
> I've noticed a bit, and talked with other marine players to confirm it wasn't a one off thing. One charger ram can give someone 7-8 bleeds, or make them bleed across their entire body. This feels very excessive.

It does overall damage instead of targeted damage. This prevents getting organ damage/fracs on one charge or stupid insta delimbs.

Unless you want the latter, this is how it's going to be.

**Comment by Carnage068** [comment]
> Made alacrit fruit work on charger, but it wanes when charging


THANK YOU.

**Comment by PhantomEpicness** [comment]
ready for merge

**Review by nauticall — APPROVED** [review]
Sprites are fine, at least the ones on this PR. If there'll be new sprites on the CM xeno side as well they'll need another looking at

File: code/datums/effects/xeno_strains/xeno_speed.dm [hunk 22]
```diff
@@ -26,6 +26,14 @@
 		return FALSE
 	return ..()
 
+/datum/effects/xeno_speed/process_mob()
+	..()
+	var/mob/living/carbon/affected_mob = affected_atom
+	if(HAS_TRAIT(affected_mob, TRAIT_CHARGING))
+		to_chat(affected_mob, SPAN_XENOWARNING("Your bowels are evacuated as you charge!"))
```
Comment (stanalbatross) [inline code/datums/effects/xeno_strains/xeno_speed.dm:33]: better message please.

File: code/datums/effects/xeno_strains/xeno_speed.dm [hunk 23]
```diff
@@ -26,6 +26,14 @@
...
+		qdel(src)
+		return FALSE
+	return ..()
```
Comment (stanalbatross) [inline code/datums/effects/xeno_strains/xeno_speed.dm:36]: newline below

**Review by stanalbatross — CHANGES_REQUESTED** [review]

**Comment by Carnage068** [comment]
Welp its back now, we did it.

**Comment by Wintermote** [comment]
> It does overall damage instead of targeted damage. This prevents getting organ damage/fracs on one charge or stupid insta delimbs.
> 
> Unless you want the latter, this is how it's going to be.

Still seems very weird. You're being hit by a blunt object, so getting bleeds on every body part just doesn't thematically fit. Like I'd expect that from a ravager ability, not the crusher.

**Comment by PhantomEpicness** [comment]
> > It does overall damage instead of targeted damage. This prevents getting organ damage/fracs on one charge or stupid insta delimbs.
> > Unless you want the latter, this is how it's going to be.
> 
> Still seems very weird. You're being hit by a blunt object, so getting bleeds on every body part just doesn't thematically fit. Like I'd expect that from a ravager ability, not the crusher.
Screenshot_419

I looked deeper and brute damage over 10 has a random chance of causing bleeding. This is a product of human damage code and reworking damage is not the scope of this PR.

What we can do however, which I will note for the eventual balance patch, is lower the limb spread chance. This will focus the damage on fewer limbs, causing less overall damage but more focused damage, which I have tried to avoid because one hit massive organ damage is not fun.

**Comment by Carnage068** [comment]
> What we can do however, which I will note for the eventual balance patch, is lower the limb spread chance. This will focus the damage on fewer limbs, causing less overall damage, and therefore less overall bleeding, but more focused damage, which I have tried to avoid because one hit massive organ damage is not fun.

Or you can just lower the charge damage a bit overall. Would probably help, considering that some people are getting one shot without being hit by a full charge, and sometimes one shot without even getting hit by charge at all.

**Comment by PhantomEpicness** [comment]
> > What we can do however, which I will note for the eventual balance patch, is lower the limb spread chance. This will focus the damage on fewer limbs, causing less overall damage, and therefore less overall bleeding, but more focused damage, which I have tried to avoid because one hit massive organ damage is not fun.
> 
> Or you can just lower the charge damage a bit overall. Would probably help, considering that some people are getting one shot without being hit by a full charge, and sometimes one shot without even getting hit by charge at all.

agreed

**Comment by c4xmaniac** [comment]
Consider making the charger crusher faster, at least getting up to full momentum. It really takes a long time, and its really easy to stop the crusher from getting up to to charge speed. Its on the cusp of being fun/good, but its not there yet imo.

**Comment by Carnage068** [comment]
> Consider making the charger crusher faster, at least getting up to full momentum. It really takes a long time, and its really easy to stop the crusher from getting up to to charge speed. Its on the cusp of being fun/good, but its not there yet imo.

It's supposed to need room to start up a full charge. Being able to bodyblock it to prevent it from getting to that speed is just called good counterplay. At the same time, it shouldn't be almost completely immobile when it IS bodyblocked though. It would just make things too binary. One of the differences between Charger and Base Crusher is that it's supposed to be faster, not just sometimes, but in general. Or at least, that's what was fun for me. Being fast is just fun.

**Pull request merged**